packages/pipeline/src/concepts/queries/radar_orderbook_usd_prices.ts

```diff
@@ -0,0 +1,51 @@
+import { Connection } from 'typeorm';
+
+/**
+ * Async function that removes the concepts.radar_orderbook_usd_prices table if it exists
+ * @param connection connection to the db instance
+ */
+export async function deleteTableAsync(connection: Connection): Promise<void> {
+    await connection.query(`drop table if exists concepts.radar_orderbook_usd_prices`);
+}
+
+/**
+ * Async function that creates the concepts.radar_orderbook_usd_prices table
+ * @param connection connection to the db instance
+ */
+export async function createTableAsync(connection: Connection): Promise<void> {
+    await connection.query(
+        `
+create table concepts.radar_orderbook_usd_prices as
+with usd_prices as (
```

Comment:
where do the tables/views `usd_prices1` and `usd_prices2` come from?

Reply:
We left join on `usd_prices` twice. `usd_prices1` is from the first join, `usd_prices2` is from the second. We join twice to check the USD price of the token on both sides of the trade. The way the logic is written, if there is a match (meaning we have the info) for the first token (the buy token), we do not join on that row again. If there is no match, we try the second token (the sell token) to see if we know its price at that time. If there is no match, we still have a row for that trade since it is a left join, but do price information (`null`).

Reply:
Ahhh I see it now. Missed that the first time. Thank you for explaining!